### PR TITLE
Bugfix/6.4.1/tdi 37328 salesforce connection validation

### PR DESCRIPTION
**Pull request by mbasiuk-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-37328

**What is the new behavior?**
When user has expired password and tries to login he will get an error message about it.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [x] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/317/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/317/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/328/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/328/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/341/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/341/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/353/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/353/)

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceSourceOrSink.java [hunk 1]
```diff
@@ -191,6 +201,26 @@ protected PartnerConnection doConnection(ConnectorConfig config, boolean openNew
...
+     * @param config connector configuration with endpoint/userId/password
+     * @param connection to be used for login in Salesforce.
+     * @throws ConnectionException if password has been expired or bad connection to Salesforce.
+     * @see com.sforce.soap.partner.PartnerConnection#PartnerConnection(ConnectorConfig config)
+     */
+    private void performLogin(ConnectorConfig config, PartnerConnection connection) throws ConnectionException {
+        config.setServiceEndpoint(config.getAuthEndpoint());
+        LoginResult loginResult = connection.login(config.getUsername(), config.getPassword());
+        if (loginResult.isPasswordExpired()) {
+            throw new ConnectionException(SalesforceRuntime.getI18nMessage("error.expiredPassword"));
```
Comment (chmyga) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceSourceOrSink.java]: Why do you use SalesforceRuntime messages, not SalesforceSourceOrSink messages?

Comment (mbasiuk-talend) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceSourceOrSink.java]: As I know we can get i18n messages only for definition part, I extended utility class SalesforceRuntime to get i18n messages for runtime part.

Comment (mbasiuk-talend) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/SalesforceSourceOrSink.java]: I agree, after discussion. Will change.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/464/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/464/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/467/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/467/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/501/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/501/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/504/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/504/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/512/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/512/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/514/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/514/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/515/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/515/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/517/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/517/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/518/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/518/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/519/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/519/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/531/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/531/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/535/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/535/)

**Pull request merged**